### PR TITLE
feat: add get starknet package

### PR DESCRIPTION
**Pull request by stanleyyconsensys**
**Description**:
the PR is to add create a compilable wallet object that bridged to metamask starknet snap as a wallet options to support the get-starknet library

Detail implementation
- create a class object named `MetaMaskSnap` to communicate with the snap via the window ethereum provider
- create a class object named `MetaMaskAccount` to extend the Account class from `starknet.js `package, and overriding execute, signMessage, declare with MetaMaskSnap
- create a class object named `MetaMaskSigner` to implement the `SignerInterface` interface from `starknet.js` package, and implement all member methods with `MetaMaskSnap`
- create a class object named `MetaMaskSnapWallet` to implement the same API as IStarknetWindowObject

### CODE REVIEW

File: packages/get-starknet/src/wallet.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+    return [this.selectedAddress];
+  }
+
+  async isPreauthorized() {
+    return true;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  on<E extends WalletEvents>() {
```
Comment (stanleyyconsensys) [inline packages/get-starknet/src/wallet.ts:137]: throw error when not support

File: packages/get-starknet/src/wallet.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    return;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  off<E extends WalletEvents>() {
+    return;
```
Comment (stanleyyconsensys) [inline packages/get-starknet/src/wallet.ts:143]: throw error when not support

File: packages/get-starknet/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+Execute the following cmd to build the project in [module federation]([https://webpack.js.org/concepts/module-federation/] ) standard
+```bash
+yarn build:fed
+```
+
+
+## How to use
+
+```javascript
+const walletInstance = new MetaMaskSnapWallet(
```
Comment (mario-christopher) [inline packages/get-starknet/README.md:16]: Can you also include the statements for importing the `MetaMaskSnapWallet()` object in this example ?

File: packages/get-starknet/index.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
+import {  MetaMaskSnap } from './src/snap';
+import {  MetaMaskSnapWallet } from './src/wallet';
+
+const init = async() => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const provider =  await MetaMaskSnap.GetProvider(window as any) as any;
+    const wallet = new MetaMaskSnapWallet(provider, '*');
+    
+    const address = await wallet.enable();
+    console.log("address", address)
```
Comment (mario-christopher) [inline packages/get-starknet/index.ts:10]: Is this `console.log()` reqd in the prod code ?

File: packages/get-starknet/index.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        params: {
+            chainId: '0x534e5f4d41494e',
+        },
+    });
+    console.log("wallet_switchStarknetChain", wallet_switchStarknetChain)
+
+    const wallet_addStarknetChain = await wallet.request( {
+        type: 'wallet_addStarknetChain',
+        params: {
+            chainId: '0x534e5f4d41494e',
```
Comment (mario-christopher) [inline packages/get-starknet/index.ts:23]: There are repeated uses of this ChainId.
Please move it to a `const`.

File: packages/get-starknet/index.ts [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    const address = await wallet.enable();
+    console.log("address", address)
+
+    const wallet_switchStarknetChain = await wallet.request( {
+        type: 'wallet_switchStarknetChain',
+        params: {
+            chainId: '0x534e5f4d41494e',
+        },
+    });
+    console.log("wallet_switchStarknetChain", wallet_switchStarknetChain)
```
Comment (mario-christopher) [inline packages/get-starknet/index.ts:18]: Is this `console.log()` reqd in the prod code ?

File: packages/get-starknet/index.ts [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    const wallet_addStarknetChain = await wallet.request( {
+        type: 'wallet_addStarknetChain',
+        params: {
+            chainId: '0x534e5f4d41494e',
+            chainName: 'StarkNet',
+            rpcUrls: ['https://stg-rpc.starknet.io'],
+            blockExplorerUrls: ['https://stg-explorer.starknet.io'],
+        },
+    });
+    console.log("wallet_addStarknetChain", wallet_addStarknetChain)
```
Comment (mario-christopher) [inline packages/get-starknet/index.ts:29]: Is this `console.log()` reqd in the prod code ?

File: packages/get-starknet/index.ts [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+                    address: "0x69b49c2cc8b16e80e86bfc5b0614a59aa8c9b601569c7b80dde04d3f3151b79",
+                    amount: "1554785",
+                }
+            ]
+        },
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } as any
+
+    const signData = await wallet.account.signMessage(message)
+    console.log("signData", signData)
```
Comment (mario-christopher) [inline packages/get-starknet/index.ts:78]: Is this `console.log()` reqd in the prod code ?

File: packages/get-starknet/public/index.html [hunk 9]
```diff
@@ -0,0 +1 @@
+<html></html>
```
Comment (mario-christopher) [inline packages/get-starknet/public/index.html:1]: What is the purpose of this HTML file ?
Can you add come comments in the Code file to describe its need and usage ?

File: packages/get-starknet/index.ts [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+        domain: {
+            name: "myDapp", 
+            version: "1",
+            chainId: "0x534e5f474f45524c49", 
+        },
+        message: {
+            id: "0x0000004f000f",
+            from: "0x2c94f628d125cd0e86eaefea735ba24c262b9a441728f63e5776661829a4066",
+            amount: "400",
+            nameGamer: "Hector26",
```
Comment (mario-christopher) [inline packages/get-starknet/index.ts:63]: Is this Test data ?
Can you add some inline code comments as to what is happening in this file ?

**Comment by mario-christopher** [comment]
Should we update the Diagram of the PR to reflect the current design ? It still shows an NPM package. Should we instead describe the Fed.Module here ?

File: packages/get-starknet/public/index.html (hunk 9 above)
Comment (stanleyyconsensys) [reply packages/get-starknet/public/index.html:1]: good catch , it should be deleted, i will remove them

File: packages/get-starknet/README.md (hunk 3 above)
Comment (stanleyyconsensys) [reply packages/get-starknet/README.md:16]: agree

Comment (stanleyyconsensys) [reply packages/get-starknet/README.md:16]: but i will refer to get-starknet v3 user guide, as our interface is implemented with get-starknet

Comment (stanleyyconsensys) [reply packages/get-starknet/README.md:16]: the package is for get-starknet, but no one else, i think we dont need a readme

File: packages/get-starknet/public/index.html (hunk 9 above)
Comment (stanleyyconsensys) [reply packages/get-starknet/public/index.html:1]: it is to test the module fed working fine or not, i will keep it

**Review by mario-christopher — APPROVED** [review]

**Pull request merged**